docs/custom-flows/add-phone.mdx

```diff
@@ -0,0 +1,155 @@
+---
+description: Learn how to use Clerk's API to build a custom flow for adding a phone number to a user's account.
+---
+
+# Build a custom flow for adding a phone number to a user's account
+
+<Callout type="danger">
+  This guide is for users who want to build a *custom* user interface using the Clerk API. To use a *prebuilt* UI, you should use Clerk's [Account Portal pages](/docs/account-portal/overview) or [prebuilt components](/docs/components/overview).
+</Callout>
+
+Users are able to add multiple phone numbers to their account.
+
+Every user has a [`User`](/docs/references/javascript/user/user) object that represents their account. The `User` object has a `phoneNumbers` property that contains all the phone numbers associated with the user.
+
+The following example demonstrates how to build a custom user interface that allows users to add a phone number to their account. The example:
+1. Uses the [`useUser()`](/docs/references/react/use-user) hook to get the `User` object.
+1. Uses the [`User.createPhoneNumber()`](/docs/references/javascript/user/create-metadata#create-phone-number) method to add the phone number to the user's account.
+  - A new [`PhoneNumber`](/docs/references/javascript/phone-number) object is created and stored in `User.phoneNumbers`.
+1. Uses the `prepareVerification()` method on the newly created `PhoneNumber` object to send a verification code to the user.
+1. Uses the `attemptVerification()` method on the same `PhoneNumber` object with the verification code provided by the user to verify the phone number.
+
+<Tabs items={["Next.js"]}>
+  <Tab>
+    ```jsx {{ filename: 'app/account/add-phone/page.tsx' }}
+    'use client';
+
+    import * as React from 'react';
+    import { useUser } from '@clerk/nextjs';
+    import { PhoneNumberResource } from '@clerk/types';
+
+    export default function Page() {
+      const { isLoaded, user } = useUser();
+      const [phone, setPhone] = React.useState('');
+      const [code, setCode] = React.useState('');
+      const [isVerifying, setIsVerifying] = React.useState(false);
+      const [successful, setSuccessful] = React.useState(false);
+      const [phoneObj, setPhoneObj] = React.useState<
+        PhoneNumberResource | undefined
+      >();
+
+      if (!isLoaded) return null;
+
+      if (isLoaded && !user?.id) {
+        return <p>You must be logged in to access this page</p>;
+      }
+
+      // Handle addition of the phone number
+      const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+
+        try {
+          // Add unverified phone number to user
+          const res = await user.createPhoneNumber({ phoneNumber: phone });
+          // Reload user to get updated User object
+          await user.reload();
+
+          // Find the phone number that was just added
+          const phoneNumber = user.phoneNumbers.find((a) => a.id === res.id);
```

Comment:
This line creates the reference. The `setPhoneObj(phoneNumber);` line is just passing that reference to `useState`.

Reply:
fixed ! [5a84226](https://github.com/clerk/clerk-docs/pull/1198/commits/5a84226fa7273bcfa42167e79e05ef2965c70e2c)